### ISSUE TITLE
Pressing both fn keys puts keyboard into numlock

### ISSUE THREAD

**Issue by chughes87** [keyboardio/Kaleidoscope#151]
The expected behavior is that the keyboard goes out of numlock mode when the fn keys are released. However it remains in numlock mode until you press the numlock button.

**Comment by algernon** [comment]
The TL;DR version of this issue is that layers are confusing, even for the firmware. The `Fn` keys are - under the hood - keys that tell the firmware to go to the next layer, whatever that is. So pressing one goes from the `QWERTY` layer to the `Fn` layer. Then pressing the other goes from `Fn` to `Numpad`. And therein lies the problem!

Because `Numpad` overrides the palm keys to turn them into keys that momentary go back to the `Fn` layer. Why is this a problem? Because when you release a key, the firmware checks what you released **on the active layer**! It will see a `Layer1` release, and will turn that layer off (`Fn`), but will keep the Numpad layer active, because it was never told to turn it off.

So the fix is to make sure that holding both `Fn` keys do not get you to the numpad layer, and we avoid all this confusion. The PR I opened against the factory firmware (see above) does just that.
